test/integration/controllers/transactions.test.js

```diff
@@ -1,17 +1,62 @@
+const StellarSdk = require('stellar-sdk');
+const transactions = require('../../../server/controllers').transactions;
+const db = require('../../../server/models');
+const { TokenType, Wallet, Transaction } = db;
+
 describe('Provenance and Redemption', () => {
-    it('reliably follow FIFO when no redemptions have occured', async (done) => {
-        // create a tokentype
-        // distribute tokens to 2 wallets
-            // each of these wallets gives tokens to wallet4
-        // check provenance from wallet 4 == wallet2
+    const INITIAL_WALLET_AMOUNT = 10000;
+    const AMOUNT = 1000;
+    let tokenType;
+    let walletOwnerKeypair;
+
+    beforeEach(async (done) => {
+        walletOwnerKeypair = StellarSdk.Keypair.random();
+        tokenType = await TokenType.create({
+            Name: 'tokenName',
+            ExpiryDate: '2020',
+            sponsor_uuid: walletOwnerKeypair.publicKey(),
+            totalTokens: 10000,
+        });
+        await Wallet.create({
+            wallet_uuid: walletOwnerKeypair.publicKey(),
+            tokentype_uuid: tokenType.uuid,
+            balance: INITIAL_WALLET_AMOUNT,
+        });
         done();
     });
-    it('reliably follow FIFO when no redemptions have occured', async (done) => {
-        // create a tokentype
-        // distribute tokens to 2 wallets (wallet2, wallet3)
-            // each of these wallets gives tokens to wallet4
-        // redeem token amount equal to first transaction (wallet3 w/ wallet4)
-        // check provenance from wallet 4 == wallet3
+
+    afterEach(async (done) => {
+        await TokenType.destroy({ where: {} });
+        await Wallet.destroy({ where: {} });
+        await Transaction.destroy({ where: {} });
         done();
     });
+
+    it('reliably gives provenance chain', async (done) => {
+        let receiver3 = StellarSdk.Keypair.random();
+        await createOriginTransaction(walletOwnerKeypair, tokenType.uuid, AMOUNT, async (tObject) =>{
```

Comment:
this is pretty confusing to read -- suggest breaking this up into variables rather than a chained function call

Reply:
I'm assuming this comment is directed at the function that begins on line 35. I will do as you have suggested.